x/evm/abci.go

```diff
@@ -197,15 +197,25 @@ func handleContractCallWithTokenToEVM(ctx sdk.Context, event types.Event, bk typ
 }
 
 func setMessageToNexus(ctx sdk.Context, n types.Nexus, event types.Event, asset *sdk.Coin) error {
+	sourceChain := funcs.MustOk(n.GetChain(ctx, event.Chain))
+
 	var message nexus.GeneralMessage
 	switch e := event.GetEvent().(type) {
 	case *types.Event_ContractCall:
+		sender := nexus.CrossChainAddress{
+			Chain:   sourceChain,
```

Comment:
forgot that cross chain address stores the chain struct instead of chain name lol